exir/emit/test/test_emit.py

```diff
@@ -812,6 +812,323 @@ def map_fn(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
         outputs = loaded_model(inputs)[0]
         torch.allclose(outputs, f(*inputs))
 
+    def test_emit_scan_basic(self) -> None:
+        """Test basic scan emission: verifies instruction structure for cumulative sum."""
+        from torch._higher_order_ops.scan import scan
+
+        class ScanCumSum(torch.nn.Module):
+            def forward(self, xs: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+                def combine_fn(carry, x):
+                    new_carry = carry + x
+                    return new_carry, new_carry.clone()
+
+                init = torch.zeros_like(xs[0])
+                return scan(combine_fn, init, xs)
+
+        f = ScanCumSum()
+        # Use contiguous tensor to avoid stride=0 issue
+        inputs = (torch.arange(15).float().reshape(5, 3),)
+
+        module = to_edge(
+            export(f, inputs, strict=True),
+            compile_config=exir.EdgeCompileConfig(_check_ir_validity=False),
+        )
+        program = module.to_executorch().executorch_program
+
+        op_table = program.execution_plan[0].operators
+        instructions = program.execution_plan[0].chains[0].instructions
+
+        # Collect all operator names in the program
```

Comment:
honestly all the ops seem like implementation details and should not be tested

Reply:
I was using it as a sort of a proxy that the general pattern was emitted. If you want we can just test the end 2 end behavior though.